sky/provision/kubernetes/utils.py

```diff
@@ -2028,6 +2195,79 @@ def get_skypilot_pods(context: Optional[str] = None) -> List[Any]:
     return pods
 
 
+def is_tpu_on_gke(accelerator: str) -> bool:
+    """Determins if the given accelerator is a TPU supported on GKE."""
+    return accelerator in GKE_TPU_ACCELERATOR_TO_GENERATION
+
+
+def get_node_accelerator_count(attribute_dict: dict) -> int:
+    """Retrieves the count of accelerators from a node's resource dictionary.
+
+    This method checks the node's allocatable resources or the accelerators
+    already deployed on the node, using pod objects that describe resource
+    requests.
+
+    Args:
+        attribute_dict: Containing resource information from a node, such as
+            allocatable or requested resources.
+
+    Returns:
+        Number of accelerators allocated or available from the node. If no
+            resource is found, it returns 0.
+    """
+    if GPU_RESOURCE_KEY in attribute_dict:
+        return int(attribute_dict[GPU_RESOURCE_KEY])
+    elif TPU_RESOURCE_KEY in attribute_dict:
+        return int(attribute_dict[TPU_RESOURCE_KEY])
+    return 0
```

Comment:
```suggestion
    if GPU_RESOURCE_KEY in attribute_dict:
        return int(attribute_dict[GPU_RESOURCE_KEY])
    if TPU_RESOURCE_KEY in attribute_dict:
        return int(attribute_dict[TPU_RESOURCE_KEY])
    return 0
```
nit. also, should we assert there is only one key in the dict?

Reply:
@cblmemo I have two questions:
1. Why do we want to update `elif` to `if`?
2. `attribute_dict` is an item returned by kubernetes api, why would we want to `assert` on a value returned by 3rd part library?

Reply:
1. It is just a nit. Align the format of two cases (both using `if`) looks better to me but feel free to keep the current way.
2. I mean `assert not (GPU_RESOURCE_KEY in attribute_dict and TPU_RESOURCE_KEY in attribute_dict)`. Reason that I think this is necessary is that, if we have both resource keys in the attribute, why are we returning the GPU resource value instead of TPU resource value? Those two values looks symmetrical in this function. If I do missed something so that both keys can present in the attribute dictionary at the same time, a comment elaborating the priority (GPU > TPU) would be helpful!

Reply:
@cblmemo Makes sense. Thanks for the elaboration. Added the assertion :)